homeassistant/components/snips.py

```diff
@@ -57,21 +68,36 @@
     vol.Optional(ATTR_SITE_ID, default='default'): str,
     vol.Optional(ATTR_CUSTOM_DATA, default=''): str
 })
-
 SERVICE_SCHEMA_SAY_ACTION = vol.Schema({
     vol.Required(ATTR_TEXT): str,
     vol.Optional(ATTR_SITE_ID, default='default'): str,
     vol.Optional(ATTR_CUSTOM_DATA, default=''): str,
     vol.Optional(ATTR_CAN_BE_ENQUEUED, default=True): cv.boolean,
     vol.Optional(ATTR_INTENT_FILTER): vol.All(cv.ensure_list),
 })
+SERVICE_SCHEMA_FEEDBACK = vol.Schema({
+    vol.Optional(ATTR_SITE_ID, default='default'): str
+})
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
+async def async_setup(hass, config):
     """Activate Snips component."""
-    @asyncio.coroutine
-    def message_received(topic, payload, qos):
+    def set_feedback(site_ids, state):
+        """Set Feedback sound state."""
+        site_ids = (site_ids if site_ids
+                    else config[DOMAIN].get(CONF_SITE_IDS))
+        topic = (FEEDBACK_ON_TOPIC if state
+                 else FEEDBACK_OFF_TOPIC)
+        for site_id in site_ids:
+            payload = json.dumps({'siteId': site_id})
+            hass.components.mqtt.async_publish(
+                FEEDBACK_ON_TOPIC, None, qos=0, retain=False)
```

Comment:
Shouldn't you clear the retained message on the off topic if setting the on topic?

Reply:
Off isn't set to be retained since that is the default if snips restarts